lib/auth/auth_with_roles.go

```diff
@@ -5078,3 +5131,11 @@ func verbsToReplaceResourceWithOrigin(stored types.ResourceWithOrigin) []string
 	}
 	return verbs
 }
+
+// createErrorChannel takes an error and produces a channel of type T as well as an error channel
+// emitting a single error. This is useful for streaming handlers when they need to return an error.
+func createErrorChannel[T any](err error) (chan T, chan error) {
```

Comment:
Can we do this without generics for now? We still support Go 1.17 on v8 so this won't backport cleanly.

Reply:
Sure thing